vue-app/src/pages/GridPage.vue

```diff
@@ -0,0 +1,345 @@
+<template>
+  <div class="page-wrapper">
+    <div class="page-content">
+      <div :style="{ width: '100%' }">
+        <div class="windows-container">
+          <grid-button-window title="Aspects">
+            <grid-config-button
+              :key="index"
+              v-for="(el, index) in allAspects"
+              @Clicked="chooseAspect(el.id)"
+              :style="{
+                textDecoration: !el.selected ? 'line-through' : 'none',
+              }"
+            >
+              {{ el.aspectText }}
+            </grid-config-button>
+          </grid-button-window>
+
+          <grid-button-window title="Grid options">
+            <grid-config-button
+              @Clicked="
+                () => {
+                  elements = getGeneratedElements();
+                }
+              "
+            >
+              Generate
+            </grid-config-button>
+
+            <grid-config-button
+              @Clicked="
+                () => {
+                  cssProps.gridAutoFlow =
+                    cssProps.gridAutoFlow == 'dense' ? 'row' : 'dense';
+                }
+              "
+            >
+              {{ cssProps.gridAutoFlow }}
+            </grid-config-button>
+            <grid-config-slider
+              _name="Element size"
+              :_value="itemSize"
+              @GSchange="(e) => changeElementSize(e)"
+            >
+            </grid-config-slider>
+            <grid-config-slider
+              _name="Gap size"
+              :_value="gapSize"
+              @GSchange="(e) => changeGapSize(e)"
+            >
+            </grid-config-slider>
+          </grid-button-window>
+        </div>
```

Comment:
this component is very long, can't we split it into shorter components?

Reply:
No. This window is just a custom wrapper with specific styling which groups all buttons and sliders into a logic group. Btw its prettier fault that the code is so padded... ugh